### DISCUSSION TITLE
Guidance Needed for Intermediate User

### DISCUSSION THREAD

**Discussion by anaradutoiu** [Lightning-AI/pytorch-lightning#7972] [Lightning Trainer API: Trainer, LightningModule, LightningDataModule]
Hello!

I have started to use PyTorch Lightning for a few weeks now. I have already went through some of the videos and introductory posts like the step-by-step walk-through tutorial.

I am know familiar to the basic concepts, however every time I see PL code written by someone more experienced I feel like I am missing out on stuff. Usually, when this happens, I go and read the documentation, look through the GitHub implementation etc. But when I try to look for PL documentation, I feel like most of it is really superficial and beginner-oriented. I can’t really find a detailed explanation of how to use the features or any link to the exact place of the GitHub implementation.

Could anybody guide me towards better understanding PL? I feel like it’s a very powerful tool and I would like to have a deeper understanding of it. Maybe I am just overseeing something, so I would appreciate any help.

**Comment by tshu-w** [comment]
I'm a newbie too. I spent two days reading through all the documentation to make sure I knew what features were available. After this I think the best way to master a tool is to read the source code, whether it is written by someone else or official.
